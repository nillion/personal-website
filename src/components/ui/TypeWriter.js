import PropTypes from 'prop-types';
import React, { Component } from 'react';

class TypeWriter extends Component {
    constructor(props) {
        super(props);
        this.state = {
            index: 0,
            displayedText: ''
        };
    }

    componentDidMount() {
        this.type();
    }

    componentWillUnmount() {
        this._timeout && clearTimeout(this._timeout);
    }

    type() {
        const { speed, text } = this.props;
        let { index } = this.state;
        if (index < text.length) {
            const displayedText = `${this.state.displayedText}${text.charAt(index)}`;
            index++;

            this.setState({ displayedText, index }, () => {
                this._timeout = setTimeout(() => {
                    this.type();
                }, speed);
            });
        }

    }

    render() {
        const { displayedText } = this.state;
        return (
            <span>{displayedText}</span>
        );
    }
}

export default TypeWriter;

TypeWriter.propTypes = {
    speed: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired
};
