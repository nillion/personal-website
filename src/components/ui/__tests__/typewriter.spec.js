import React from 'react';
import renderer from 'react-test-renderer';

import TypeWriter from '../TypeWriter';

describe("TypeWriter", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(<TypeWriter text="Test" speed={100} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})