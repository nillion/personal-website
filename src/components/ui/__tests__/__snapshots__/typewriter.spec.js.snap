// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeWriter renders correctly 1`] = `
<span>
  T
</span>
`;
